lib/views/question/question_container_view.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:kayla_flutter_ic/utils/route_paths.dart';
+import 'package:kayla_flutter_ic/views/question/question_container_state.dart';
+import 'package:kayla_flutter_ic/views/question/question_container_view_model.dart';
+import 'package:kayla_flutter_ic/views/question/question_view.dart';
+
+final questionViewModelProvider = StateNotifierProvider.autoDispose<
+    QuestionContainerViewModel, QuestionContainerState>(
+  (ref) {
+    return QuestionContainerViewModel();
+  },
+);
+
+class QuestionContainerView extends ConsumerStatefulWidget {
+  final String surveyId;
+  final int questionNumber;
+
+  const QuestionContainerView({
+    super.key,
+    required this.surveyId,
+    required this.questionNumber,
+  });
+
+  @override
+  QuestionContainerViewState createState() => QuestionContainerViewState();
+}
+
+class QuestionContainerViewState extends ConsumerState<QuestionContainerView> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _setUpData();
+    return QuestionView(
+      child: Container(),
+      onNextQuestion: () => _nextQuestion(),
+      onSubmit: () => _submit(),
+    );
+  }
+
+  void _setUpData() {
+    ref.read(questionViewModelProvider.notifier).bindData(
+          surveyId: _getSurveyId(),
+          questionNumber: _getQuestionNumber(),
+        );
+  }
+
+  void _nextQuestion() {
+    try {
+      Map<String, String> params = <String, String>{};
+      params[RoutePath.surveyDetail.pathParam] = _getSurveyId();
+
+      final nextQuestionNumber = _getQuestionNumber() + 1;
+      Map<String, String> queryParams = <String, String>{};
+      queryParams[RoutePath.question.queryParams.first] =
+          nextQuestionNumber.toString();
+      context.goNamed(
+        RoutePath.question.name,
+        params: params,
+        queryParams: queryParams,
+      );
+    } catch (error) {
+      // ignore: avoid_print
+      print(error);
+    }
+  }
+
+  String _getSurveyId() {
+    final arguments =
+        ModalRoute.of(context)?.settings.arguments as Map<String, String>;
+    return arguments[RoutePath.question.pathParam] ?? '';
+  }
+
+  int _getQuestionNumber() {
+    final arguments =
+        ModalRoute.of(context)?.settings.arguments as Map<String, String>;
+    return int.parse(arguments[RoutePath.question.queryParams.first] ?? '0');
```

Comment:
we can store the current question in VM too

Reply:
The idea is to not calculate any logic in View part. The View part should only render UI and execute command based on VM's call